tests/baseline/tsfresh_features_test.py

```diff
@@ -0,0 +1,417 @@
+            timeseries_str = str(timeseries_json).replace('{\'results\': ', '').replace('}', '')
+
+        full_timeseries = eval(timeseries_str)
+        timeseries = full_timeseries[:60]
+        self.assertEqual(int(timeseries[0][0]), 1369677886)
+        self.assertEqual(len(timeseries), 60)
+
+        for ts, value in timeseries:
+            metric = 'tsfresh_features_test'
+            timestamp = int(ts)
+            value = str(float(value))
+            utc_ts_line = '%s,%s,%s\n' % (metric, str(timestamp), value)
+            with open(tmp_csv, 'a') as fh:
+                fh.write(utc_ts_line)
+
+        if os.path.isfile(tmp_csv):
+            file_created = True
+        else:
+            file_created = False
+        self.assertEqual(file_created, True)
+
+        df_features = None
+        if os.path.isfile(t_fname_out):
+            return True
+        df = pd.read_csv(tmp_csv, delimiter=',', header=None, names=['metric', 'timestamp', 'value'])
+        df.columns = ['metric', 'timestamp', 'value']
+        df_features = extract_features(df, column_id='metric', column_sort='timestamp', column_kind=None, column_value=None)
+
+        df_created = None
+        df_assert = None
+        # Test the DataFrame
+        try:
+            df_created = str(df_features.head())
+            if df_created:
+                df_assert = True
+                self.assertEqual(df_assert, True)
+        # Catch when df_created is None
+        except AttributeError:
+            self.assertEqual(df_created, True)
+            pass
+        # Catch if not defined
+        except NameError:
+            self.assertEqual(df_created, True)
+            pass
+
+        if os.path.isfile(t_fname_out):
+            return True
+
+        # Write to disk
+        df_features.to_csv(fname_out)
+
+        if os.path.isfile(fname_out):
+            file_created = True
+        else:
+            file_created = False
+        self.assertEqual(file_created, True)
+
+        if os.path.isfile(t_fname_out):
+            return True
+
+        # Transpose
+        df_t = None
+        df_t = df_features.transpose()
+        # df_t.sort_values(by=[0], ascending=[False])
+
+        # Test the DataFrame
+        df_t_created = None
+        df_t_assert = None
+        try:
+            df_t_created = str(df_t.head())
+            if df_t_created:
+                df_t_assert = True
+                self.assertEqual(df_t_assert, True)
+        # Catch when df_t_created is None
+        except AttributeError:
+            self.assertEqual(df_t_created, True)
+            pass
+        # Catch if not defined
+        except NameError:
+            self.assertEqual(df_t_created, True)
+            pass
+
+        # Write the transposed csv
+        df_t.to_csv(t_fname_out)
+
+        if os.path.isfile(t_fname_out):
+            file_created = True
+        else:
+            file_created = False
+        self.assertEqual(file_created, True)
+        return True
```

Comment:
It may be simpler.  The only reason there is a transposed csv is because I am a human and as such it is easier for me to look at the data vertical rather than horizontally.  I realise panda probably does not care either way, column wise or row wise.

It probably is achievable in a more simple manner, if you do not have to rewrite it that way and figure it out all yourself.

And the results of all of that would be exactly the same.  I do not want to spend a lot more time on this at this stage.  I may be happy to try and refactor minimalistic perfection at some point in the future, however I want to move back to getting on with developing my application asap.